### ISSUE TITLE
Support transitive membership lookup in google_cloud_identity_group_memberships

### ISSUE THREAD

**Issue by carlpett** [hashicorp/terraform-provider-google#11675]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description
Currently `google_cloud_identity_group_memberships` only returns direct memberships. In larger organisations, it is common to build trees of groups-in-groups so eg dev@domain has members dev-us@domain and dev-eu@domain which in turn may have subgroups, etc.

What we want is a listing of all members, regardless of group membership depth. The Cloud Identity APIs used in the data source today already support this by way of `GroupsMembershipsService.SearchTransitiveGroups`.

### New or Affected Resource(s)

* `google_cloud_identity_group_memberships`

### Potential Terraform Configuration

```tf
data "google_cloud_identity_group_memberships" "members" {
  group = "groups/123eab45c6defghi"
  transitive = true # Or something along those lines
}
```

### References

Docs: https://pkg.go.dev/google.golang.org/api/cloudidentity/v1#GroupsMembershipsService.SearchTransitiveGroups
Also noted there is that this is only supported on higher tiers of Google Workspace, so some warning in the docs may be in order.

### Other
We'd be willing to contribute the code, but I'm unsure if this needs to go through the magician? Also debatable whether to do this by flag in the existing data source, or a if a new one is better.

**Comment by carlpett** [comment]
@rileykarson Is there a way for us to contribute this? If so, we'd be happy to, but would need some help with the questions above.

**Comment by roaks3** [comment]
Yes, contributions can be made using documentation at https://googlecloudplatform.github.io/magic-modules/. For this, it might be worth copying an existing data source like `DataSourceGoogleCloudIdentityGroupMemberships` and working off of that.

**Comment by jbbqqf** [comment]
Hi! Triaging older Cloud Identity requests — this one is shipped, just under a separate data source.

Rather than adding a `transitive` flag to `google_cloud_identity_group_memberships`, the maintainers shipped a dedicated data source: [`google_cloud_identity_group_transitive_memberships`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/cloud_identity_group_transitive_memberships). Source lives at [`google/services/cloudidentity/data_source_cloud_identity_group_transitive_memberships.go`](https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/cloudidentity/data_source_cloud_identity_group_transitive_memberships.go) and wraps the same `SearchTransitiveGroups` API call from the issue body.

Usage:

```hcl
data "google_cloud_identity_group_transitive_memberships" "members" {
  group = "groups/123eab45c6defghi"
}
```

(Note: as you flagged in the original request, transitive lookups require a higher Google Workspace tier — this is a server-side limit, not a Terraform restriction.)

Would you mind closing this out? If the data source is missing a field you need (e.g. relation labels), point me at it and I'll dig.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the data source presence was verified manually against the current `terraform-provider-google` `main` branch.)
